[extensions][ts] 1/n Update to Model Parser impl

Ankush-lastmile opened the pull request:
[extensions][ts] 1/n Update to Model Parser impl

## What

Copied over ts implementation from core library


modelparser class name
- `HuggingFaceTextGenerationModelParser`
model.id() ->
- HuggingFaceTextGenerationModelParser


## Why

Starting diff for stack to create an extension that serves as an example on how to write a model parser extension including comprehensive docs, docstrings, etc

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/314).
* __->__ #314
* #312

rossdanlm commented:
nit: rename `HuggingFaceTextGenerationModelParser` to `HuggingFaceTextGenerationModelParserExtension`

rossdanlm commented on extensions/HuggingFace/typescript/hf.ts:
```diff
@@ -148,20 +205,33 @@ export class HuggingFaceTextGenerationModelParser extends aiconfig.Parameterized
+      output = constructOutput(response);
     }
+
+    prompt.outputs = [output];
```
Why do we do this here and not before?

Ankush-lastmile replied:
Most of these changes are from [PR#283](https://github.com/lastmile-ai/aiconfig/pull/283)

rossdanlm commented on extensions/HuggingFace/typescript/hf.ts:
```diff
@@ -178,7 +248,8 @@ export class HuggingFaceTextGenerationModelParser extends aiconfig.Parameterized
 
     if (output.output_type === "execute_result") {
-      return output.data as string;
+      return (output.data as TextGenerationOutput | TextGenerationStreamOutput)
+        .generated_text as string;
```
This makes sense to me. I feel we should change the core lib too? Although tbh most code will hit the L242 anyways and never reach here

Ankush-lastmile replied:
I'm assuming you were referring to similar discussion as here: https://github.com/lastmile-ai/aiconfig/pull/283#discussion_r1399248692

I'll update the python to be similar, sounds like we want to keep the response structure as much as possible

rossdanlm reviewed (approved):
Generally lgtm, left comments for questions

Ankush-lastmile commented:
>nit: rename HuggingFaceTextGenerationModelParser to HuggingFaceTextGenerationModelParserExtension

Great suggestion, Thanks!

renamed `HuggingFaceTextGenerationModelParser` to `HuggingFaceTextGenerationModelParserExtension`

updated pr description.
